### PR TITLE
CLN: Remove 0.7 deprecations

### PR DESCRIPTION
**Pull request by bashtage**
Remove all 0.7 scheduled deprecations
Rmove tests that use deprecated features

### CODE REVIEW

File: statsmodels/tools/tools.py [hunk 1]
```diff
@@ -383,23 +383,6 @@ def clean0(matrix):
     return np.array(np.transpose(val))
 
 
-def rank(X, cond=1.0e-12):
```
Comment (josef-pkt) [inline statsmodels/tools/tools.py]: there is no internal use of this anymore?
we might have silently changed the threshold cond at some point when switching to numpy default

**Comment by josef-pkt** [comment]
looks fine to me for 0.9

cross-link for threshold in matrix_rank, pinv and similar #2628

File: statsmodels/tools/tools.py (hunk 1 above)
Comment (bashtage) [inline statsmodels/tools/tools.py]: Only place I found it was in a test.

**Comment by josef-pkt** [comment]
@bashtage  Thank you for cleaning up, merged

**Pull request merged**